File: Julia/AoC/src/Math.jl
module Math

export append_digits

"""
  Append two Ints like this:
  ```julia
  append_digits(102, 4050) => 1024050
  ```
"""
function append_digits(a::Int, b::Int)::Int
    n = b
    while 0 < n
        a *= 10
        n = div(n, 10)
    end
    a + b
end

end

File: Swift/Sources/Utils/math.swift
//
//  math.swift
//  aoc
//

/// Append two Ints
///
/// ## example
///
/// ```swift
/// append_digits(234, 67) => 23467
/// ```
public func append_digits(_ a: Int, _ b: Int) -> Int {
    func step(_ n: Int, _ x: Int) -> Int {
        if x < 10 {
            return n * 10 + b
        }
        return step(n * 10, x / 10)
    }
    return step(a, b)
}
